src/MemoryPack.Core/MemoryPackWriter.cs

```diff
@@ -166,12 +166,15 @@ public void WriteString(string? value)
         }
 
         var copyByteCount = value.Length * 2;
```

Comment:
Why not directly check `value.Length` ? 😄 

```csharp
if(value.Length > 0)
{
    //....
}
else
{
    //....
}
```

Reply:
I was thinking getting the ```dest``` first by using the pre-defined temporary variable copyByteCount

Reply:
otherwise its a bit unneat:
```csharp
if(value.Length > 0)
{
  ref var dest = ref GetSpanReference(4);
  Unsafe.WriteUnaligned(ref dest, 0);
  Advance(4);
}
else
{
  var copyByteCount = value.Length * 2;
  ref var dest = ref GetSpanReference(copyByteCount + 4);
  Unsafe.WriteUnaligned(ref dest, value.Length);
  Unsafe.CopyBlockUnaligned(ref Unsafe.Add(ref dest, 4), ref src, (uint)copyByteCount);
  Advance(copyByteCount + 4);
}
```
and obviously by inspection, we need to define ```dest``` anyway, and to define ```dest``` we need ```copyByteCount```, therefore I took the calculation of ```copyByteCount``` and the definition of the variable ```dest``` out of the if-else block

Reply:
Well. If the owner thinks it's worth optimizing here with empty string, I suggest writing a single code block for it.

```csharp

if(value is null)
{
    //...
    return;
}

if(value.Length < 1)
{
    //...
    return;
}

//...

return;

```

Reply:
the length of the string has to be written as long as the string is not empty, otherwise it is impossible to deserialise lol

but I think we can optimised the empty string by having a if check and skip the cast char* to byte* and the unnecessary copy haha